File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/core/BaseActivity.java
package com.tai.androidtai.modules.core;

import android.os.Bundle;

import com.tai.androidtai.AndroidTaiApplication;
import com.tai.androidtai.dagger.components.AndroidTaiComponent;
import com.tai.androidtai.dagger.modules.ActivityModule;

import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity implements BaseContract.View {

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    protected ActivityModule getActivityModule() {
        return new ActivityModule(this);
    }

    protected AndroidTaiComponent getApplicationComponent() {
        return ((AndroidTaiApplication) getApplication()).getApplicationComponent();
    }

}

File: AndroidTai/app/src/main/java/com/tai/androidtai/modules/core/BaseActivity.kt
package com.tai.androidtai.modules.core

import android.os.Bundle

import com.tai.androidtai.AndroidTaiApplication
import com.tai.androidtai.dagger.components.AndroidTaiComponent
import com.tai.androidtai.dagger.modules.ActivityModule

import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity(), BaseContract.View {

    protected val activityModule: ActivityModule
        get() = ActivityModule(this)

    protected val applicationComponent: AndroidTaiComponent
        get() = (application as AndroidTaiApplication).applicationComponent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

}
